src/test/java/org/agenda/repo/AppointmentCacheRepositoryTest.java

```diff
@@ -45,7 +46,20 @@ void create_happyFlow() {
         assertEquals(appointmentId, appointment.getAppointmentId());
         assertEquals(startDateTime, appointment.getStartDateTime());
         assertEquals(endDateTime, appointment.getEndDateTime());
-        assertEquals(1, appointmentCacheRepository.getAppointmentList().size());
+        assertEquals(1, appointmentCacheRepository.getList().size());
+    }
+
+    @Test
+    void create_shouldThrowIllegalArgumentException_appointmentAlreadyExists() {
```

Comment:
Ik zou geen IllegalArgumentException gooien. Maar eerder een eigen `CreationException`:

Bijv: `CreationException extends RuntimeException`

Het is namelijk een valide argument.